infra/cifuzz/docker.py

```diff
@@ -31,7 +32,9 @@
     '-e', 'FUZZING_ENGINE=' + constants.DEFAULT_ENGINE, '-e', 'CIFUZZ=True'
 ]
 
-EXTERNAL_PROJECT_IMAGE = 'external-project'
+UNIQUE_ID_SUFFIX = '-' + uuid.uuid4().hex
+
+EXTERNAL_PROJECT_IMAGE = 'external-cfl-project' + UNIQUE_ID_SUFFIX
```

Comment:
Hmm, how does this work across process boundaries? i.e. build_fuzzers vs run_fuzzers being run in a separate process. Would this break those cases? 

Does this just work for GitLab because there's a single entrypoint? 

To solve this for split build/run_fuzzers, perhaps we can use something like `GITHUB_RUN_ID` / one of the runner IDs in https://docs.gitlab.com/ee/ci/variables/predefined_variables.html, etc for all the CI systems.

Reply:
We actually don't really need to know the name of the image in run_fuzzers. The only thing we use the name for is deleting the image (we should really do this in the previous step).
I don't like the id of using run-id because it's more platform specific API that integrators will need to support.

Reply:
Ah right, we just run the binaries in a separate image. 

Is not deleting the image enough of a blocker to fix that in this PR first before merging?  Otherwise this LGTM.

Reply:
I wouldn't say so. Especially because this only affects CFL and not CIFuzz.

Reply:
The approach I was gonna go with was to have some kind of statefile that can be used for cross process communication and for machine parsing of results.

Reply:
Done

Reply:
Done